scripts/run_tox_envs.py

```diff
@@ -0,0 +1,53 @@
+import os
+from qiskit_versions import *
+
+versions_info = get_qiskit_versions_info()
+
+# Create a tox env for each qiskit version
+for info in versions_info:
+    # Since Python 3.6 and below have reached end of life
+    # Using 3.7 here as it is compatible with all qiskit-terra versions
+    python_version = "3.7"
+    qiskit_version = info["version"]
+    env_name = "q_v" + qiskit_version
+ 
+    tox_config = """
+[tox]
+minversion = 3.7
+envlist = {}
+
+[testenv]
+usedevelop = True
+
+deps =
+qiskit-terra=={}
+requests
+pyzx
+
+commands = 
+    python {}/scripts/circuit_depth_and_gate_count.py
+""".format(env_name, qiskit_version, "{toxinidir}")
```

Comment:
Consider using f-strings. Something like this:
```suggestion
    tox_config = f"""
[tox]
minversion = 3.7
envlist = {env_name}

[testenv]
usedevelop = True

deps =
qiskit-terra=={qiskit_version}
requests
pyzx

commands = 
    python {toxinidir}/scripts/circuit_depth_and_gate_count.py
"""
```